app/harvesters/nypl_harvester.rb

```diff
@@ -0,0 +1,249 @@
+##
+# A harvester for NYPL's API
+#
+class NyplHarvester
+  include Krikri::Harvester
+
+  DEFAULT_URI = 'http://api.repo.nypl.org/api/v1'
+  DEFAULT_NAME = 'nypl'
+  DEFAULT_BATCHSIZE = 10
+  DEFAULT_THREAD_COUNT = 5
+
+  ##
+  # Initialize, and set default options as appropriate.
+  #
+  # @param opts [Hash] a hash of options as defined by {.expected_opts}
+  #
+  # @example
+  #    Typical instantiation, good for most cases:
+  #
+  #        NyplHarvester.new(nypl: { apikey: '[key]' })
+  #
+  # Parameters to 'opts':
+  #
+  # - uri:        See Krikri::Harvester#initialize.
```

Comment:
`:uri` and `:name` should not be included in `#expected_opts` (see [here](https://github.com/dpla/KriKri/blob/develop/lib/krikri/harvester.rb#L150-L181))